File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.js
// 모든 분-초로 되어있는 시간을 초로 바꾸기
// prev -> 10초를 빼주고, pos가 10 이하일 경우 0으로 바꾸기
// next -> 10초를 더해주고, 전체 길이 - pos가 10 이하일 경우 전체길이로 바꾸기
// 오프닝 체크는 명령을 수행하기 전/후로 둘다 체크하기
const changeTime = (time) => {
  const [m, s] = time.split(":")
  return +m*60 + +s
}

// 오프닝 시작, 오프닝 끝, 현재 위치 비교
const checkOpening = (s, e, c) => {
  if(s <= c && c <= e) return e
  return c
}

function solution(video_len, pos, op_start, op_end, commands) {
  video_len = changeTime(video_len)
  pos = changeTime(pos)
  op_start = changeTime(op_start)
  op_end = changeTime(op_end)
  
  commands.forEach((cmd) => {
    pos = checkOpening(op_start, op_end, pos);

    if(cmd === "prev"){
      pos -= 10
      if(pos < 10) pos = 0
    } else {
      pos += 10
      if(video_len - pos < 10) pos = video_len
    }

    pos = checkOpening(op_start, op_end, pos);
  })

  const [result_m, result_s] = [Math.floor(pos / 60).toString().padStart(2, "0"), (pos%60).toString().padStart(2, "0")]
  return result_m + ":" + result_s
}

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
# 시간 초로 변환
def changeTime(time):
  m, s = map(int, time.split(":"))
  return 60*m + s

# 오프닝 구간 체크는 이동하기 전과 이동하고 나서 두 번 해야함
def checkOpening(s, e, c):
  if s <= c <= e:
    return e
  return c

def solution(video_len, pos, op_start, op_end, commands):
  # 전부 초로 변환
  video_len = changeTime(video_len)
  pos = changeTime(pos)
  op_start = changeTime(op_start)
  op_end = changeTime(op_end)

  for cmd in commands:
    pos = checkOpening(op_start, op_end, pos)

    # 앞으로 가기
    if cmd == "prev":
      pos -= 10
      if pos < 10:
        pos = 0
    # 뒤로 가기
    if cmd == "next":
      pos += 10
      if video_len - pos < 10:
        pos = video_len

    pos = checkOpening(op_start, op_end, pos)
  
  result_m, result_s = divmod(pos, 60)
  return f"{result_m:02d}:{result_s:02d}"